test.sh

```diff
@@ -95,8 +95,9 @@ function test_rest_api() {
     # Without this, the DB could be locked because
     # the migrations have not yet finished.
     sleep 2
-    for i in {1..5}; do
-        populate_db_with_mock_data 2>/dev/null && break || sleep 1
+    for _ in {1..5}; do
+        populate_db_with_mock_data 2>/dev/null && break
+        sleep 1
```

Comment:
I guess that the meaning of both versions is equivalent, but the previous one reported [`SC2015: Note that A && B || C is not if-then-else. C may run when A is true.`](https://www.shellcheck.net/wiki/SC2015)